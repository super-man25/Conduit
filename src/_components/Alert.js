// @flow
import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styled from 'styled-components';

import { colors, shadows, zIndexes } from '_constants';
import { actions } from '_state/alert';
import { Icon } from './Icon';
import { Flex } from './Flex';

const StyledApiAlert: React.ComponentType<{}> = styled(Flex)`
  position: absolute;
  right: 25px;
  top: 25px;
  z-index: ${zIndexes.BASE};
  width: 350px;
  background-color: ${colors.white};
  border-color: ${(props) =>
    props.type === 'apiError' ? colors.red : colors.green};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
  cursor: pointer;
  padding: 25px;
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  margin-right: 15px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
  margin-bottom: 3px;
`;

const MsgText: React.ComponentType<{}> = styled.div`
  font-size: 13px;
  font-weight: 300;
`;

export const Alert = () => {
  const alertState = useSelector(({ alert }) => alert);
  const dispatch = useDispatch();
  const clearAlert = () => dispatch(actions.clear());

  const show = alertState.type !== null && alertState.message !== null;
  const statusColor =
    alertState.type === 'apiError' ? colors.red : colors.green;
  const statusText = alertState.type === 'apiError' ? 'Error:' : 'Success!';

  return (
    <StyledApiAlert {...alertState} show={show} onClick={clearAlert}>
      <StatusIcon>
        <Icon name={alertState.type} size={40} color={statusColor} />
      </StatusIcon>
      <Flex direction="column">
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </Flex>
    </StyledApiAlert>
  );
};
