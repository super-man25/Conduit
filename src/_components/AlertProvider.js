import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { history } from '_helpers';
import { actions as alertActions } from '_state/alert';

class AlertProviderPresenter extends React.Component {
  constructor(props) {
    super(props);

    const { dispatch } = this.props;
    history.listen(() => {
      dispatch(alertActions.clear());
    });
  }

  render() {
    const { alert, children } = this.props;

    // TODO: style alerts
    return (
      <div>
        {alert &&
          alert.show && (
            <div className={`alert ${alert.type}`}>{alert.message}</div>
          )}
        {children}
      </div>
    );
  }
}

AlertProviderPresenter.propTypes = {
  /** Redux dispatch */
  dispatch: PropTypes.func,

  /** Alert object */
  alert: PropTypes.shape({
    message: PropTypes.string,
    type: PropTypes.string
  }),

  /** Children */
  children: PropTypes.node
};

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const AlertProvider = connect(mapStateToProps)(AlertProviderPresenter);
