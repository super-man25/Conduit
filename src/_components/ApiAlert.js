// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { cssConstants, shadows, zIndexes } from '_constants';
import { Icon } from './Icon';

type Props = {
  alertState: {
    type: 'api-error' | 'api-success',
    message: string
  }
};

export const ApiAlertDiv: React.ComponentType<{}> = styled.div`
  position: absolute;
  right: 24px;
  top: 24px;
  z-index: ${zIndexes.BASE};
  font-size: 18px;
  width: 357px;
  height: 90px;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-color: ${(props) =>
    props.type === 'api-error'
      ? cssConstants.SECONDARY_RED
      : cssConstants.SECONDARY_GREEN};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 26px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 93px;
  font-size: 1.2em;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
`;

const MsgText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 54px;
  left: 93px;
  font-size: 0.7em;
  font-weight: 300;
  margin-right: 24px;
  color: ${cssConstants.PRIMARY_BLACK};
`;

export class ApiAlertPresenter extends React.Component<Props> {
  render() {
    const { alertState } = this.props;
    const show = {
      show: alertState.type !== null && alertState.message !== null
    };
    const statusColor =
      alertState.type === 'api-error'
        ? cssConstants.SECONDARY_RED
        : cssConstants.SECONDARY_GREEN;
    const statusText = alertState.type === 'api-error' ? 'Error:' : 'Success!';

    return (
      <ApiAlertDiv {...alertState} {...show}>
        <StatusIcon>
          <Icon name={alertState.type} size={39} color={statusColor} />
        </StatusIcon>
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </ApiAlertDiv>
    );
  }
}

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const ApiAlert = connect(
  mapStateToProps,
  null
)(ApiAlertPresenter);
