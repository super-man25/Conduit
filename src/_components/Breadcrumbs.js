import styled from 'styled-components';
import { cssConstants } from '../_constants';

export const Breadcrumbs = styled.div`
  width: 100%;
  height: 30px;
  margin: 0;
  padding: 0;
  background: ${cssConstants.PRIMARY_WHITE};
  font-size: 12px;
`;
