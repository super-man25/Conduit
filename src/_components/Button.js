// @flow

import * as React from 'react';
import styled from 'styled-components';
import { cssConstants, shadows } from '_constants';
import { withBoxModelProps } from '_helpers/style-utils';

type Props = {
  fontSize: string,
  textAlign: string,
  minWidth: string,
};

export const Button: React.ComponentType<Props> = withBoxModelProps(styled.button`
  border: none;
  min-width: ${(props) => props.minWidth || '100px'};
  padding: 1rem 2rem;
  text-align: ${(props) => props.textAlign || 'center'};
  border-radius: 3px;
  font-size: ${(props) => props.fontSize || '1rem'};
  cursor: pointer;
  transition-duration: 0.3s;

  &:disabled {
    cursor: not-allowed;
  }
`);

export const PrimaryButton = styled(Button)`
  transition: opacity 100ms ease-out;
  color: ${cssConstants.PRIMARY_WHITE};
  background-color: ${cssConstants.PRIMARY_BLUE};

  &:disabled {
    color: ${cssConstants.PRIMARY_GRAY};
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    background-color: ${cssConstants.PRIMARY_BLUE_HOVER};
  }
`;

export const SecondaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 2px solid ${cssConstants.PRIMARY_BLUE};
  color: ${cssConstants.PRIMARY_BLUE};
  transition: border linear;

  &:disabled {
    border: 2px solid ${cssConstants.PRIMARY_GRAY};
    color: ${cssConstants.PRIMARY_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    outline: 2px solid ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    border: 3px solid ${cssConstants.PRIMARY_BLUE};
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_BLUE};
  }
`;

export const TertiaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
  opacity: 0.8;

  &:disabled {
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    opacity: 1;
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  }
`;

export const TextButton = styled(Button)`
  color: ${cssConstants.PRIMARY_BLUE};
  background-color: inherit;

  &:disabled {
    color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
  }
`;
