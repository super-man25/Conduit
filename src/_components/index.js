export * from './Alert';
export * from './AsyncButton';
export * from './Breadcrumbs';
export * from './Button';
export * from './Center';
export * from './CenteredLoader';
export * from './ChartContainer';
export * from './ChartLegend';
export * from './Checkbox';
export * from './ChipButton';
export * from './ContentWrapper';
export * from './DatePicker';
export * from './DateRangeDropdown';
export * from './DateRangeFilter';
export * from './Dropdown';
export * from './DropdownButton';
export * from './EDLink';
export * from './EDText';
export * from './Flex';
export * from './Form';
export * from './FormInput';
export * from './FormInputArray';
export * from './HelpBlockDiv';
export * from './HorizontalBars';
export * from './Icon';
export * from './Icon/SVGIcons';
export * from './ImageLayout';
export * from './Input';
export * from './Integration';
export * from './Layout';
export * from './LeftNav';
export * from './LegendItem';
export * from './Loader';
export * from './MailtoLink';
export * from './MainContent';
export * from './Modal';
export * from './Panel';
export * from './ReportDownloadButton';
export * from './ScreenTitleBlock';
export * from './ScrollableList';
export * from './SecuredRoute';
export * from './SelectBox';
export * from './SelectDropdown';
export * from './Setting';
export * from './SettingButton';
export * from './SiteHeader';
export * from './SortableButton';
export * from './Spacing';
export * from './StyledTags';
export * from './Table';
export * from './Toggle';
export * from './UserWelcome';
