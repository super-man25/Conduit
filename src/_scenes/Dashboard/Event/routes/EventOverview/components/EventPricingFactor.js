import React, { Component, Fragment } from 'react';
import {
  NumberInputField,
  Input,
  Text,
  ApiAlert,
  EDTextButton,
  Flex
} from '_components';
import { connect } from 'react-redux';
import { actions } from '_state/event';
import { formatUSD } from '_helpers/string-utils';

export class EventPricingFactorPresenter extends Component {
  state = {
    isEditing: false,
    modifier: this.props.modifier,
    modelInvalid: false
  };

  formatDisplayUnit = (unit, value) => {
    return unit === '$'
      ? `${formatUSD(value || 0)}`
      : `${(value || 0).toFixed(2)}${unit}`;
  };

  getDisplayType = (type) => {
    return type === 'eventScoreModifier' ? 'Event Score' : 'Spring';
  };

  getDisplay = (props) => {
    const { type, unit, base, modifier } = props;

    return `${this.getDisplayType(type)}: ${this.formatDisplayUnit(
      unit,
      (base || 0) + modifier
    )}`;
  };

  getPlaceholderText = (type) => {
    return type === 'eventScoreModifier'
      ? 'Event Score Modifier'
      : 'Spring Modifier';
  };

  toggleEdit = () => {
    this.setState({
      isEditing: !this.state.isEditing,
      modifier: this.props.modifier,
      modelInvalid: false
    });
  };

  calculateResult = (unit, base) => {
    const { modifier } = this.state;
    const result = parseFloat(base || 0) + parseFloat(modifier);
    return this.formatDisplayUnit(unit, result);
  };

  onBlur = (event) => {
    if (!event.target.value) {
      this.setState({
        modifier: '0'
      });
    }
  };

  update = (event) => {
    this.setState({
      modifier: event.target.value,
      modelInvalid: false
    });
  };

  save = () => {
    const {
      saveAdminModifiers,
      seasonId,
      event: {
        event: { id, eventScoreModifier, springModifier }
      },
      type
    } = this.props;

    const payload = {
      eventId: id,
      eventScoreModifier,
      springModifier,
      [type]: parseFloat(this.state.modifier).toFixed(2)
    };
    saveAdminModifiers(
      payload.eventId,
      payload.eventScoreModifier,
      payload.springModifier,
      seasonId,
      (error) => {
        error ? this.setState({ modelInvalid: true }) : this.toggleEdit();
      }
    );
  };

  render() {
    const { type, unit, base } = this.props;
    return (
      <Fragment>
        <ApiAlert />
        {this.state.isEditing ? (
          <Fragment>
            <Flex justify="space-between">
              <Text marginBottom=".33rem" size={15}>
                {this.getDisplayType(type)}
              </Text>
              <Flex>
                <EDTextButton
                  type="primary"
                  size="medium"
                  weight="normal"
                  onClick={this.toggleEdit}
                  marginRight="10px"
                >
                  CANCEL
                </EDTextButton>
                <EDTextButton
                  type="primary"
                  size="medium"
                  weight="normal"
                  onClick={this.save}
                >
                  SAVE
                </EDTextButton>
              </Flex>
            </Flex>
            <Flex align="center">
              <Text marginRight="8px" size={18}>
                {this.formatDisplayUnit(unit, base)} +{' '}
              </Text>
              <NumberInputField
                component={Input}
                type="number"
                value={this.state.modifier}
                width="40%"
                onChange={this.update}
                onBlur={this.onBlur}
                placeholder={this.getPlaceholderText(type)}
                inValid={this.state.modelInvalid}
              />
              <Text marginLeft="8px" size={18}>
                = {this.calculateResult(unit, base)}
              </Text>
            </Flex>
          </Fragment>
        ) : (
          <Flex justify="space-between">
            <Text size={18}>{this.getDisplay(this.props)}</Text>
            <EDTextButton
              type="primary"
              size="medium"
              weight="normal"
              onClick={this.toggleEdit}
            >
              EDIT
            </EDTextButton>
          </Flex>
        )}
      </Fragment>
    );
  }
}

const mapStateToProps = (state) => ({
  event: state.event,
  savingAdminModifiers: state.savingAdminModifiers,
  seasonId: state.season.activeId
});

const mapDispatchToProps = {
  saveAdminModifiers: actions.saveAdminModifiers
};

export const EventPricingFactor = connect(
  mapStateToProps,
  mapDispatchToProps
)(EventPricingFactorPresenter);
