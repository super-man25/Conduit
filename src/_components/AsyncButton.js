// @flow

import * as React from 'react';
import { cssConstants } from '_constants';
import { PrimaryButton, Loader } from '_components';

type Props = {
  isLoading: boolean,
  disabled?: boolean,
  children: React.Node,
};

export const AsyncButton = (props: Props) => {
  const { isLoading, children, ...rest } = props;
  return (
    <PrimaryButton {...rest}>
      {props.isLoading ? (
        <Loader small color={cssConstants.PRIMARY_WHITE} />
      ) : (
        props.children
      )}
    </PrimaryButton>
  );
};
