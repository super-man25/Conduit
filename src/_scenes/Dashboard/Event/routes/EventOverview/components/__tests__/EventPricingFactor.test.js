import React from 'react';
import { shallow, mount } from 'enzyme';
import { EDTextButton, NumberInputField } from '_components';
import { EventPricingFactorPresenter } from '../EventPricingFactor';
import configureStore from 'redux-mock-store';

describe('<EventPricingFactor />', () => {
  const initialState = {
    type: null,
    message: null
  };
  const mockStore = configureStore();
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });

  const eventScoreProps = {
    type: 'eventScoreModifier',
    unit: '$',
    base: 1,
    modifier: 1.5
  };
  const springProps = {
    type: 'springModifier',
    unit: '%',
    base: 1,
    modifier: 0.5
  };

  it('renders correctly for event score', () => {
    const wrapper = shallow(
      <EventPricingFactorPresenter {...eventScoreProps} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('isEditing')).toBe(false);
    expect(wrapper.state('modifier')).toBe(eventScoreProps.modifier);
  });

  it('renders correctly for spring', () => {
    const wrapper = shallow(<EventPricingFactorPresenter {...springProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('isEditing')).toBe(false);
    expect(wrapper.state('modifier')).toBe(springProps.modifier);
  });

  it('can update event score modifier value on change', () => {
    const wrapper = shallow(
      <EventPricingFactorPresenter {...eventScoreProps} store />
    );
    const spy = jest.spyOn(wrapper.instance(), 'update');
    const event = { target: { value: 1.5 } };
    wrapper.find(EDTextButton).simulate('click');
    wrapper
      .find(NumberInputField)
      .at(0)
      .simulate('change', event);
    expect(wrapper.state('modifier')).toBe(event.target.value);
    expect(spy).toHaveBeenCalled();
  });

  it('can update spring modifier value on change', () => {
    const wrapper = shallow(
      <EventPricingFactorPresenter {...springProps} store />
    );
    const spy = jest.spyOn(wrapper.instance(), 'update');
    const event = { target: { value: -1.5 } };
    wrapper.find(EDTextButton).simulate('click');
    wrapper
      .find(NumberInputField)
      .at(0)
      .simulate('change', event);
    expect(wrapper.state('modifier')).toBe(event.target.value);
    expect(spy).toHaveBeenCalled();
  });
});
