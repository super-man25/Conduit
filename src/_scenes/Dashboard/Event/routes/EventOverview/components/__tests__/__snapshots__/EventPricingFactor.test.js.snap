// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventPricingFactor /> renders correctly for event score 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventPricingFactorPresenter
    base={1}
    modifier={1.5}
    type="eventScoreModifier"
    unit=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(ApiAlertPresenter) />,
        <Flex
          justify="space-between"
        >
          <styled.p
            size={18}
          >
            Event Score: 2.500
          </styled.p>
          <Styled(styled.div)
            onClick={[Function]}
            size="medium"
            type="primary"
            weight="normal"
          >
            EDIT
          </Styled(styled.div)>
        </Flex>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.p
              size={18}
            >
              Event Score: 2.500
            </styled.p>,
            <Styled(styled.div)
              onClick={[Function]}
              size="medium"
              type="primary"
              weight="normal"
            >
              EDIT
            </Styled(styled.div)>,
          ],
          "justify": "space-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Event Score: 2.500",
              "size": 18,
            },
            "ref": null,
            "rendered": "Event Score: 2.500",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "EDIT",
              "onClick": [Function],
              "size": "medium",
              "type": "primary",
              "weight": "normal",
            },
            "ref": null,
            "rendered": "EDIT",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(ApiAlertPresenter) />,
          <Flex
            justify="space-between"
          >
            <styled.p
              size={18}
            >
              Event Score: 2.500
            </styled.p>
            <Styled(styled.div)
              onClick={[Function]}
              size="medium"
              type="primary"
              weight="normal"
            >
              EDIT
            </Styled(styled.div)>
          </Flex>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.p
                size={18}
              >
                Event Score: 2.500
              </styled.p>,
              <Styled(styled.div)
                onClick={[Function]}
                size="medium"
                type="primary"
                weight="normal"
              >
                EDIT
              </Styled(styled.div)>,
            ],
            "justify": "space-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Event Score: 2.500",
                "size": 18,
              },
              "ref": null,
              "rendered": "Event Score: 2.500",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "EDIT",
                "onClick": [Function],
                "size": "medium",
                "type": "primary",
                "weight": "normal",
              },
              "ref": null,
              "rendered": "EDIT",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<EventPricingFactor /> renders correctly for spring 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventPricingFactorPresenter
    base={1}
    modifier={0.5}
    type="springModifier"
    unit="%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(ApiAlertPresenter) />,
        <Flex
          justify="space-between"
        >
          <styled.p
            size={18}
          >
            Spring: 1.50%
          </styled.p>
          <Styled(styled.div)
            onClick={[Function]}
            size="medium"
            type="primary"
            weight="normal"
          >
            EDIT
          </Styled(styled.div)>
        </Flex>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.p
              size={18}
            >
              Spring: 1.50%
            </styled.p>,
            <Styled(styled.div)
              onClick={[Function]}
              size="medium"
              type="primary"
              weight="normal"
            >
              EDIT
            </Styled(styled.div)>,
          ],
          "justify": "space-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Spring: 1.50%",
              "size": 18,
            },
            "ref": null,
            "rendered": "Spring: 1.50%",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "EDIT",
              "onClick": [Function],
              "size": "medium",
              "type": "primary",
              "weight": "normal",
            },
            "ref": null,
            "rendered": "EDIT",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(ApiAlertPresenter) />,
          <Flex
            justify="space-between"
          >
            <styled.p
              size={18}
            >
              Spring: 1.50%
            </styled.p>
            <Styled(styled.div)
              onClick={[Function]}
              size="medium"
              type="primary"
              weight="normal"
            >
              EDIT
            </Styled(styled.div)>
          </Flex>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.p
                size={18}
              >
                Spring: 1.50%
              </styled.p>,
              <Styled(styled.div)
                onClick={[Function]}
                size="medium"
                type="primary"
                weight="normal"
              >
                EDIT
              </Styled(styled.div)>,
            ],
            "justify": "space-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Spring: 1.50%",
                "size": 18,
              },
              "ref": null,
              "rendered": "Spring: 1.50%",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "EDIT",
                "onClick": [Function],
                "size": "medium",
                "type": "primary",
                "weight": "normal",
              },
              "ref": null,
              "rendered": "EDIT",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
